feat(curriculum): add security and privacy lessons (#68206)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lecture-understanding-security-and-privacy-in-web-applications/6a3b86e9f96f6bbcd0ed8c45.md

---
id: 6a3b86e9f96f6bbcd0ed8c45
title: What Is HTTPS and Why Is It Important?
challengeType: 19
dashedName: what-is-https-and-why-is-it-important
---

# --description--

HTTPS stands for Hypertext Transfer Protocol Secure. It is the secure version of HTTP.

As you've learned in a few previous lectures, HTTP transfers data between your browser and a website. Standard HTTP sends this data in plain text, so anyone can intercept that plain text data.

> For this reason, HTTPS uses a cryptographic protocol called Transport Layer Security (TLS) to secure this data transfer. TLS encrypts the communication channel between two points. Other systems used Secure Sockets Layer (SSL), which many web hosting platforms still use today.

Reviewer: ```suggestion
For this reason, HTTPS uses a cryptographic protocol called Transport Layer Security (TLS) to secure this data transfer. TLS encrypts the communication channel between two points. Older systems used Secure Sockets Layer (SSL), which many web hosting platforms still use today.
```

Encryption scrambles that data into unreadable code that only the user's browser and the web server hold the keys to unscramble.

For example, when you type a password into a website, HTTPS scrambles it instantly and protects credit card information, personal addresses, login credentials, and other sensitive information. Hackers who intercept the data see only random characters, so they cannot use the stolen information.

This protection is vital on public Wi-Fi networks. Open Wi-Fi networks at coffee shops, airports, and other public places lack basic security, giving cybercriminals the ability to monitor unencrypted traffic on these networks. HTTPS creates a private tunnel through the public network to keep your browsing history safe from nearby bad actors.

HTTPS also provides authentication, which proves that a website is legitimate. To use HTTPS, a website owner must obtain a digital certificate that a trusted organization issues. The certificate verifies the true identity of the website owner.

> When you visit a website and the address starts with only "http", that's one way to know that the website owner has not obtained that certificate, making the website unsafe. Modern browsers also warn you about such websites and load them more slowly.

Reviewer: Removed the line about slowly loading the page. This can be confusing as browsers do not artificially throttle http.

```suggestion
When you visit a website and the address starts with only "http", that's one way to know that the website owner has not obtained that certificate, making the website unsafe. Modern browsers also warn you about such websites.
```

---
id: 6a3b86e9f96f6bbcd0ed8c45

> title: What is HTTPS and why is it important?

Reviewer 2: ```suggestion
title: What Is HTTPS and Why Is It Important?
```

challengeType: 19
dashedName: what-is-https-and-why-is-it-important
---

# --description--

HTTPS stands for Hypertext Transfer Protocol Secure. It is the secure version of HTTP.

> As you've learnt in a few previous lectures, HTTP transfers data between your browser and a website. Standard HTTP sends this data in plain text, so anyone can intercept that plain text data.

Reviewer 2: ```suggestion
As you've learned in a few previous lectures, HTTP transfers data between your browser and a website. Standard HTTP sends this data in plain text, so anyone can intercept that plain text data.
```

Encryption scrambles that data into unreadable code that only the user's browser and the web server hold the keys to unscramble.

For example, when you type a password into a website, HTTPS scrambles it instantly and protects credit card information, personal addresses, login credentials, and other sensitive information. Hackers who intercept the data see only random characters, so they cannot use the stolen information.

This protection is vital on public Wi-Fi networks. Open Wi-Fi networks at coffee shops, airports, and other public places lack basic security, giving cybercriminals the ability to monitor unencrypted traffic on these networks. HTTPS creates a private tunnel through the public network to keep your browsing history safe from nearby bad actors.

HTTPS also provides authentication, which proves that a website is legitimate. To use HTTPS, a website owner must obtain a digital certificate that a trusted organization issues. The certificate verifies the true identity of the website owner.

> When you visit a website and the address starts with only "http", that's one way to know that the website owner has not obtained that certificate, making the website unsafe. Modern browsers also warn you about such website and loads it more slowly.

Reviewer 2: ```suggestion
When you visit a website and the address starts with only "http", that's one way to know that the website owner has not obtained that certificate, making the website unsafe. Modern browsers also warn you about such websites and load them more slowly.
```

Lastly, data integrity is another key benefit of HTTPS. This means no one can alter your data during transit. A hacker cannot inject malicious code into the website files while they travel to your device. You receive the exact data that the website owner sent.

Search engines like Google prioritize HTTPS websites, as safe websites rank higher in search results. Modern browsers also require HTTPS for advanced features such as geolocation, push notifications, and web cameras.

# --questions--

## --text--

What is the primary purpose of the digital certificate required for HTTPS websites?

## --answers--

To prove the true identity of the website owner.

---

To speed up the loading time of web pages.

### --feedback--

The certificate is issued by a trusted organization and verifies legitimacy.

---

To store the user's browsing history securely.

### --feedback--

The certificate is issued by a trusted organization and verifies legitimacy.

---

To encrypt all files stored on the server.

### --feedback--

The certificate is issued by a trusted organization and verifies legitimacy.

## --video-solution--

1

---

> 

Reviewer 2: ```suggestion
```

## --text--

What cryptographic protocol does HTTPS use to secure the transfer of data between a browser and a web server?

## --answers--

Secure Sockets Layer (SSL) only.

### --feedback--

The protocol encrypts the communication channel and is the modern standard.

---

Hypertext Transfer Protocol (HTTP).

### --feedback--

The protocol encrypts the communication channel and is the modern standard.

---

Transport Layer Security (TLS).

---

File Transfer Protocol (FTP).

### --feedback--

The protocol encrypts the communication channel and is the modern standard.

## --video-solution--

3

---

> 

Reviewer 2: ```suggestion
```
